projects/step-frontend/src/lib/modules/timeseries/discover/ts-discover.component.ts

```diff
@@ -0,0 +1,103 @@
+import { AfterViewInit, Component, inject, Inject, OnChanges, OnInit, ViewEncapsulation } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { DateFormat, MeasurementsStats, TimeSeriesService } from '@exense/step-core';
+import { HttpClient } from '@angular/common/http';
+import { DiscoverDialogData } from './discover-dialog-data';
+import { MatTableDataSource } from '@angular/material/table';
+
+@Component({
+  selector: 'step-ts-discover',
+  templateUrl: './ts-discover.component.html',
+  styleUrls: ['./ts-discover.component.scss'],
+})
+export class TsDiscoverComponent implements OnInit {
+  readonly DateFormat = DateFormat;
+  readonly PAGE_SIZE = 20;
+  readonly _dialogRef = inject(MatDialogRef);
+  dataSource: MatTableDataSource<any> = new MatTableDataSource();
+  oqlFilter: string;
+  skip = 0;
+  hasMore = true;
+  isLoading = true;
+  allColumns: string[] = [];
+  dynamicColumns: string[] = [];
+  visibleTableColumns: string[] = [];
+  stats: MeasurementsStats | undefined;
+  allSelected = true;
+  columnsSelection: { [key: string]: boolean } = {};
+
+  constructor(
+    private _matDialogRef: MatDialogRef<TsDiscoverComponent>,
+    @Inject(MAT_DIALOG_DATA) public _data: DiscoverDialogData,
+    private timeSeriesService: TimeSeriesService,
+    private http: HttpClient
+  ) {
+    this.oqlFilter = _data.oqlFilter;
+  }
+
+  ngOnInit(): void {
+    this.fetchStats();
+    this.fetchMeasurements();
+  }
+
+  private fetchMeasurements() {
+    this.isLoading = true;
+    const oql = this._data.oqlFilter;
+    this.http
+      .get<any[]>(`/rest/time-series/raw-measurements?filter=${oql}&limit=${this.PAGE_SIZE}&skip=${this.skip}`)
```

Comment:
don't we have a proper type for these measurements, can we avoid using ```any```?

Reply:
sorry, this was left off.